superset/models/core.py

```diff
@@ -670,8 +670,8 @@ def safe_sqlalchemy_uri(self) -> str:
 
     @property
     def inspector(self) -> Inspector:
-        engine = self._get_sqla_engine()
-        return sqla.inspect(engine)
+        with self.get_sqla_engine_with_context() as engine:
+            return sqla.inspect(engine)
```

Comment:
I think this will fail and the inspector will have a reference to a closed engine; we need to refactor it.